.functions.php

```diff
@@ -35,3 +29,51 @@ function checkIfAdmin() {
     return ($results[0]['admin'] == 1 || $results[0]['schedco'] == 1);
   }
 }
+
+function getUser($sessionID, $connection){
+  if(isset($sessionID)){
+    $connection->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+  
+    if(!isset($dname)) {
+      $dname = 'ambulanc_web';
+    }
+  
+    $connection->exec("USE `$dname`");
+  
+    $statement = $connection->prepare('SELECT * FROM members LEFT JOIN sessions on sessions.userID = members.id WHERE sessions.sessionID = :sessionID');
+    $statement->bindParam(":sessionID", $sessionID);
+    $statement->execute();
+    $results=$statement->fetchAll(PDO::FETCH_ASSOC);
+    updateSession($sessionID, $connection);
+    return json_encode($results[0]);
+  }
+}
+
+function updateSession($sessionID, $connection){
+  if(!isset($dname)) {
+    $dname = 'ambulanc_web';
+  }
+  $connection->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+  $stmt = $connection->prepare('SELECT * FROM `sessions` WHERE sessionID = :sessionID');
+  $stmt->bindParam(":sessionID", $sessionID);
+  $stmt->execute();
+  $current_session = $stmt->fetch();
+  date_default_timezone_set('America/New_York');
+  $current_date = new DateTime();
+  $expiration = strtotime($current_session['expiration']);
+  if (time() < $expiration){
+    $current_date = $current_date->modify('+5 day');
+    $current_date = $current_date->format('Y-m-d H:i:s');
+    $stmt = $connection->prepare('UPDATE sessions SET expiration=:expiration WHERE sessionID=:sessionID');
+    $stmt->bindParam(":sessionID",$sessionID);
+    $stmt->bindParam(":expiration", $current_date);
+    $stmt->execute();
+    setcookie("RPIA-SESSION", $sessionID, strtotime("+5 day", time()), "/");
+  }else{
+    $stmt = $connection->prepare('DELETE FROM `sessions` WHERE sessionID=:sessionID');
```

Comment:
You should change this to just be a `DELETE FROM `sessions` WHERE expiration <= CURRENT_TIMESTAMP` (whatever the equivalent is within MySQL as if a cookie expires) that happens at the top of any call to `updateSession` (the site does not operate at a scale where there's a worry about this query). Otherwise, you'll end up not properly cleaning up expired sessions where the cookie expired (and was deleted) before they visited the site.

You can keep this if statement at least to delete the cookie, but in a normal flow, it looks like the cookie and session should expire at the same time and be kept in lockstep such that there should never be a case where the user has a cookie and there is no DB session, unless they were doing some manual cookie adjustments at which the deletion is somewhat pointless as they'll just manually change the date anyway.

Reply:
This makes sense, I guess it's the niche case when they edit the cookie but it's not like it'll do anything for them besides breaking their experience. This has been resolved.